### PR TITLE
don't return err if block already in db. overwrite it

### PR DESCRIPTION
**Pull request by y0sher**

### CODE REVIEW

**Review by almogdepaz — CHANGES_REQUESTED** [review]
this is a big decision we decided this was an error for a reason, we should talk about it,
if we decide we need this this is not the place to put it,
from a database point of view this should stay an error

**Comment by y0sher** [comment]
I don't see where in the code we treat that error. until now we've treated it like we've failed to add the block and it's not in the db which is wrong. (this is what this pr fixes) I also don't see a reason why this is not the place. we can return a special error and start propagating it upwards so we can treat it but currently, I don't see any use case in our code which requires that feature. Also all the unit tests have passed without changing anything so I don't see a reason why we need that.

**Comment by zalmen** [comment]
We shouldn't ignore cases where we add a block that already exists, since it is probably a bug and we should report such cases. On the other hand, returning an error is wrong for two reasons -
1.  The caller of `AddBlock` doesn't care that the block was already there and it has nothing to do with this information.
2. If you return an error the caller might falsely assume that the operation failed and that the block isn't in the db although it is there.
This is what we should do -
Before adding a block we should check if it is already there. If it is there, print an error and don't return an error.

**Comment by y0sher** [comment]
I've changed the code to print a warning when we try to add a block that exists, but we'll not return an error. essentially doing the same effect as before apart from writing a new value to the db.
we now check the db everytime before writing, when there's no block we have a read+write. when there's a block we have 1 read.
with overwriting we always have one write.

File: mesh/meshdb.go [hunk 1]
```diff
@@ -71,7 +71,8 @@ func (m *MeshDB) Close() {
 
 func (m *MeshDB) AddBlock(bl *types.Block) error {
 	if _, err := m.getMiniBlockBytes(bl.ID()); err == nil {
-		return errors.New(fmt.Sprintf("block %v already exists in database", bl.ID()))
+		log.Warning(fmt.Sprintf("block %v already exists in database", bl.ID()))
```
Comment (zalmen) [inline mesh/meshdb.go]: change to error

**Comment by almogdepaz** [comment]
i think the solution i suggested is much better, overwriting things in the db is just the wrong way to solve this and in my opinion would result in bigger issues later on

File: mesh/meshdb.go (hunk 1 above)
Comment (y0sher) [reply mesh/meshdb.go]: done, though I find it weird to print an error but return from the function as there was no error

**Comment by y0sher** [comment]
probably not going to get in @almogdepaz is addressing it in another pr

**Pull request closed without merging**